src/server/mod.rs

```diff
@@ -457,21 +457,24 @@ impl LanguageServer for LspServer {
 
         match self.store.get(&key) {
             Ok(value) => {
-                let mut contents = Cow::Borrowed(&value);
-                for change in params.content_changes {
-                    contents = Cow::Owned(
+                // The way the spec reads, if given a list of changes to make, these changes
+                // are made in the order that they are provided, e.g. an straight iteration,
+                // applying each one as given, is the correct process. That means a change later
+                // in the list could overwrite a change made earlier in the list.
```

Comment:
Thanks for the explanation! This is helpful

Reply:
Heh, I wrote the original code, but when I went back to figure out how to refactor it, I had to go look it up in the spec again. That meant I needed to leave a comment this time.